docs/PSR7-Usage.md

```diff
@@ -0,0 +1,168 @@
+### PSR-7 Usage
+
+All PSR-7 applications comply with these interfaces 
+They were created to establish a standard between middleware implementations.
+
+> `RequestInterface`, `ServerRequestInterface`, `ResponseInterface` extend `MessageInterface`  because the `Request` and the `Response` are `HTTP Messages`.
+> When using `ServerRequestInterface`, both `RequestInterface` and `Psr\Http\Message\MessageInterface` methods are considered.
+
+
+Enough with the talking, let's put things in practice.
+The following examples will illustrate how basic operations are done in PSR-7.
+
+#####Examples (using Zend Diactoros)
+
+Zend Diactoros is an implementation for PSR-7 interfaces. It will be used to illustrate these examples.
+Installation guide for Zend Diactoros: [Zend Diactoros Documentation - Installation](https://zendframework.github.io/zend-diactoros/install/)
```

Comment:
I'm not sure we should use Diactoros here, as that sounds like an endorsement of a specific implementation. I'd change this to indicate that instantiation of messages will vary based on implementation, and leave it at that. The rest of the examples will work fine, and _should_ work the same across implementations.